frontend/src/tests/import-export.test.js

```diff
@@ -2,47 +2,226 @@
+    }
+
+    /**
+     * Given correct object data, object type, and correct object schema,
+     * carry out tests to check whether exporting correct object data to file
+     * will produce correct object data file.
+     *
+     * @param {[object]} objectData
+     * @param {string} objectType
+     * @param {object} objectSchema
+     * @param {string} fileType
+     * @returns
+     */
+    function testExportToFile(objectData, objectType, objectSchema) {
+        // export object data a CSV File object
+        const fileCSV = getObjectDataFile(
+            objectType,
+            objectData,
+            objectSchema,
+            "csv"
+        );
+        // test csv content
+        csvContainsData(fileCSV, objectData, objectSchema, objectType);
+
+        // export object data to a JSON File object
+        const fileJSON = getObjectDataFile(
+            objectType,
+            objectData,
+            objectSchema,
+            "json"
+        );
+        // test json content
+        jsonContainsData(fileJSON, objectData, objectSchema, objectType);
+    }
 
-    const instructorData = [
-        {
-            id: 2,
-            first_name: "Gordon",
-            last_name: "Smith",
-            email: "a@a.com",
-            utorid: "booger",
-        },
-        {
-            id: 3,
-            first_name: "Tommy",
-            last_name: "Smith",
-            email: "a@b.com",
-            utorid: "food",
-        },
-        {
-            first_name: "Grandpa",
-            last_name: "Boobie",
-            email: "a@d.com",
-            utorid: "fooc",
-        },
-    ];
+    /**
+     * Given correct object data, object type, correct object schema, and file type,
+     * carry out tests to check whether importing from corresponding file containing correct corresponding object data
+     * will produce correct object data.
+     *
+     * @param {[object]} objectData
+     * @param {string} objectType
+     * @param {object} objectSchema
+     * @param {string} fileType
+     * @param {string} correctness
+     * @returns
+     */
+    function testImportFromFile(
+        objectData,
+        objectType,
+        objectSchema,
+        fileType,
+        correctness = ""
+    ) {
+        if (fileType == "json") {
+            // import object data from a JSON file
+            const resultJSON = normalizeImport(
+                {
+                    fileType: "json",
+                    data:
+                        objectJSON[`${correctness}${objectType}`][
+                            `${objectType}s`
+                        ],
+                },
+                objectSchema
+            );
+            expect(resultJSON).toEqual(objectData);
+        } else if (fileType == "spreadsheet") {
+            // import object data from a CSV File
+            const workbook1 = XLSX.readFile(
+                __dirname +
+                    `/test-data-collection/${correctness}${objectType}s.csv`
```

Comment:
Yuck...it's not a good idea to do string manipulation when looking for files...it's better to hard-code the file names.